File: HelpSoftware/Help/Help.EF/Entitys/BookletArticle.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace Help.EF
{
  public class BookletArticle : IHelpEntity
  {
    [Key]
    public virtual int ID_BookletArticle { get; set; }

    public virtual Booklet Book { get; set; }

    public virtual Article Article { get; set; }

    public virtual DateTime AssingTime { get; set; }

    [Timestamp]
    public virtual byte[] TimeStamp { get; set; }

    public string GetEntityName()
    {
      return "BookletArticle";
    }

    public int GetID()
    {
      return ID_BookletArticle;
    }

    public bool GetWriteHistoryEntry()
    {
      return true;
    }
  }
}

File: HelpSoftware/Help/Help.Web/Web/src/app/EF/BookletArticle.ts
import { Booklet, Article } from "./EF";

export class BookletArticle {
  ID_BookletArticle: Number;
  Book: Booklet;
  Article: Article;
  AssingTime: Date;

  deserialize(input): BookletArticle {
    this.ID_BookletArticle = input.ID_BookletArticle;
    this.Book = new Booklet().deserialize(input.Booklet);
    this.Article = new Article().deserialize(input.Article);
    this.AssingTime = input.AssingTime;
    return this;
  }
}
